plugins_src/autouv/wpc_snap_win.erl

```diff
@@ -0,0 +1,759 @@
+    Units = [{dx, {0.0,?HUGE}},{dy, {0.0,?HUGE}}],
+    Flags = [{initial, [Ix,Iy]}],
+    wings_drag:drag_only(ScaleFun, Units, Flags, St).
+
+move(Op, St) when Op =:= center_x; Op =:= center_y; Op =:= center ->
+    S = get(?MODULE),
+    case Op of
+      center_x -> put(?MODULE, S#s{tx=0.0});
+      center_y -> put(?MODULE, S#s{ty=0.0});
+      center -> put(?MODULE, S#s{tx=0.0,ty=0.0})
+    end,
+    St;
+move(Op, St) ->
+    #s{tx=Ix,ty=Iy} = get(?MODULE),
+    MoveFun = fun([X,Y]) ->
+		      State = get(?MODULE),
+		      case Op of
+			  x ->    put(?MODULE, State#s{tx=-X});
+			  y ->    put(?MODULE, State#s{ty=-Y});
+			  free -> put(?MODULE, State#s{tx=-X,ty=-Y})
+		      end,
+		      ok
+	      end,
+    Units = [{dx,{-?HUGE,?HUGE}},{dy,{-?HUGE,?HUGE}}],
+    Flags = [{initial,[-Ix,-Iy]}],
+    wings_drag:drag_only(MoveFun, Units, Flags, St).
+
+rotate(Rotate, St) ->
+    State=get(?MODULE),
+    put(?MODULE, State#s{r=Rotate}),
+    St.
+%    RotateFun = fun({finish,_}, Dlo) -> Dlo;
+%		   ([R], Dlo) ->
+%		       State = get(?MODULE),
+%		       put(?MODULE, State#s{r=R}),
+%		       Dlo
+%		end,
+%
+%    Tvs   = {general, [{find_a_id(St), RotateFun}]},
+%    Units = [{rx, {-360.0,360.0}}],
+%    Flags = [{initial, [R0]}],
+%    wings_drag:drag_only(RotateFun, Units, Flags, St).
+
+fit(Op, St) ->
+    #s{w=IW,h=IH}=S = get(?MODULE),
+    {_,_,W,H} = wings_wm:viewport(),
+    {X,Y} = scale(W,H,IW,IH),
+    case Op of
+      x -> put(?MODULE, S#s{sx=1.0/X,tx=0.0});
+      y -> put(?MODULE, S#s{sy=1.0/Y,ty=0.0});
+      both -> put(?MODULE, S#s{sx=1.0/X,sy=1.0/Y,tx=0.0,ty=0.0})
+    end,
+    St.
+
+opacity(Opacity, St) ->
+    State=get(?MODULE),
+    wings_pref:set_value(snap_opacity, Opacity),
+    put(?MODULE, State#s{opacity=Opacity}),
+    St.
+
+%find_a_id(#st{shapes=Shs}) ->
+%    Ida = [Id || #we{id=Id,perm=Perm} <- gb_trees:values(Shs),
+%	   ?IS_VISIBLE(Perm)],
+%    Id = case length(Ida) of
+%	     0 -> wpa:error_msg(?__(1,"Visible object required."));
+%	     _ -> lists:min(Ida)
+%	 end,
+%    Id.
+
+draw_image(Image,_St) ->
+    gl:pushAttrib(?GL_ALL_ATTRIB_BITS),
+    gl:matrixMode(?GL_PROJECTION),
+    gl:loadIdentity(),
+    glu:ortho2D(0.0, 1.0, 0.0, 1.0),
+    gl:matrixMode(?GL_MODELVIEW),
+    gl:loadIdentity(),
+
+    gl:disable(?GL_LIGHTING),
+    gl:disable(?GL_DEPTH_TEST),
+    gl:disable(?GL_ALPHA_TEST),
+
+    gl:enable(?GL_BLEND),
+    gl:blendFunc(?GL_SRC_ALPHA, ?GL_ONE_MINUS_SRC_ALPHA),
+    gl:enable(?GL_TEXTURE_2D),
+    gl:bindTexture(?GL_TEXTURE_2D, Image),
+    gl:texEnvi(?GL_TEXTURE_ENV, ?GL_TEXTURE_ENV_MODE, ?GL_MODULATE),
+
+    #s{w=IW,h=IH,sx=Sx,sy=Sy,tx=Tx,ty=Ty,r=Rot,opacity=Opa} = get(?MODULE),
+    gl:'begin'(?GL_QUADS),
+    gl:color4f(1.0, 1.0, 1.0, Opa),   %%Semitransparant
+    {_,_,W,H} = wings_wm:viewport(),
+    {Xs,Ys,Xe,Ye} = {0.0,0.0,1.0,1.0},
+
+    {X,Y} = scale(W,H,IW,IH),
+
+    Center = 0.5,
+    Size   = 1.0,
+    Xrange = (X*Size)/2,
+    Yrange = (Y*Size)/2,
+
```

Comment:
We don't use intermediate mode anymore, I have already fixed this in the wpc_snap.erl file you deleted.
If you have rebased to a new master.